src/liballoc/borrow.rs

```diff
@@ -408,9 +408,7 @@ impl<'a> AddAssign<&'a str> for Cow<'a, str> {
     fn add_assign(&mut self, rhs: &'a str) {
         if self.is_empty() {
             *self = Cow::Borrowed(rhs)
-        } else if rhs.is_empty() {
-            return;
-        } else {
+        } else if !rhs.is_empty() {
```

Comment:
IMO, this makes things more confusing. Previously we were explicitly handling the three "cases": I am empty, you are empty, and one of us is not empty.

(Also applies to other change in this file).